### PR TITLE
chore(deps): Switch back to upstream openssl-src

### PR DESCRIPTION
**Pull request by jszwedko**
Since there is a new release with engine fixes.

Signed-off-by: Jesse Szwedko <jesse.szwedko@datadoghq.com>

closes: #13695

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -363,10 +363,6 @@ leveldb-sys = { git = "https://github.com/vectordotdev/leveldb-sys.git", branch
 chrono = { git = "https://github.com/vectordotdev/chrono.git", branch = "no-default-time-v0.4.22-1" }
 # Adds `Status::is_reset` test, remove after the next version is released.
 h2 = { git = "https://github.com/hyperium/h2.git", rev = "f6aa3be6719270cd7b4094ee1940751b5f4ec88e" }
-# Enables OpenSSL ENGINE support during cross compilation to musl
-# TODO remove this branch and instead use the openssl-src feature flag once it is available
-#    (see https://github.com/vectordotdev/vector/issues/13695)
-openssl-src = { git = "https://github.com/vectordotdev/openssl-src-rs.git", branch  = "enable_engine" }
```
Comment (neuronull) [inline Cargo.toml:369]: I don't think the release contains the changes we need ...

I think we need this change:

https://github.com/alexcrichton/openssl-src-rs/pull/151/files

Comment (neuronull) [reply Cargo.toml:369]: And then to add `force-engine` feature flag to our `Cargo.toml`

Comment (jszwedko) [reply Cargo.toml:369]: Ah, doh, right, I do have to add the feature flag. I think this release _does_ include the changes though. It corresponds to https://github.com/alexcrichton/openssl-src-rs/commit/b5a61821d62f4161f5480e542601668ece0e26c2 which has your commit in its history. I might be missing something though.

Comment (jszwedko) [reply Cargo.toml:369]: Added feature flag in [04188b4](https://github.com/vectordotdev/vector/pull/15375/commits/04188b4a6f2fca19caaea88c2d67b2b50b541e8f)

Comment (neuronull) [reply Cargo.toml:369]: Oh huh. Tracking the releases of this project is HARD :laughing: 

I guess if CI passes after adding that flag, that proves the changes are there.

On that note though... it looks like the musl cross build failed :sob:

File: Cargo.toml [hunk 2]
```diff
@@ -345,6 +345,7 @@ nix = { version = "0.26.1", default-features = false, features = ["socket", "sig
 [build-dependencies]
 prost-build = { version = "0.11.4", default-features = false, optional = true }
 tonic-build = { version = "0.8", default-features = false, features = ["transport", "prost"], optional = true }
+openssl-src = { version = "111", default-features = false, features = ["force-engine"] }
```
Comment (jszwedko) [inline Cargo.toml:348]: This doesn't seem to be taking effect.

Comment (jszwedko) [reply Cargo.toml:348]: J/k it is now.

**Review by neuronull — APPROVED** [review]
Really glad to see the loop closed on this one. Thanks! :tada:

**Comment by neuronull** [comment]
closes: #13695

**Pull request merged**